USAGE.md

```diff
@@ -0,0 +1,213 @@
+# SMI Metrics API
+
+The `smi-metrics` api follows the format of the official [Metrics API](https://github.com/kubernetes-incubator/metrics-server) being built on top of kubernetes to get metrics about pods and nodes but here the metrics are oriented towards the inbound and outbound requests i.e their Golden Metrics like p99_latency, P55_latency, success_count, etc for a particular workload.
+
+## Working
+
+The SMI metrics api is a Kubernetes [APIService](https://kubernetes.io/docs/tasks/access-kubernetes-api/setup-extension-api-server/) more info [here](https://github.com/deislabs/smi-metrics/blob/94dec57fdabc680cc60e4961db1707609f6b81ed/chart/templates/apiservice.yaml#L5), which is a way of extending the Kubernetes API.
+
+We will perform installation of the SMI Metrics API w.r.t linkerd. Make sure linkerd is installed and is running as per the instructions [here](https://linkerd.io/2/getting-started/), This API can be installed by running the following command
+```
+helm template chart -f dev.yaml -f linkerd.yaml --name dev
+```
+
+The APIService first informs the kubernetes API about itself and the resource types that it exposes. The following command can used to see the resource types that SMI exposes.
+
+```
+kubectl api-resources | grep smi
+NAME                              SHORTNAMES   APIGROUP                       NAMESPACED   KIND
+daemonsets                                     metrics.smi-spec.io            true         TrafficMetrics
+deployments                                    metrics.smi-spec.io            true         TrafficMetrics
+namespaces                                     metrics.smi-spec.io            false        TrafficMetrics
+pods                                           metrics.smi-spec.io            true         TrafficMetrics
+statefulsets                                   metrics.smi-spec.io            true         TrafficMetrics
+```
+
+This means that, the APIService can be queried regarding the above mentioned resource types.
+
+Now that the SMI APIService is installing, metric queries can be done through the kubernetes API.
+To Access the Kubernetes API, we can use the `proxy` command present in `kubectl`
+```
+kubectl proxy --port=8087 &
+```
+Now the Kubernetes API can be accessed at port `8087`. As Linkerd also attaches proxies to the linkerd control-plane components, we should be able to query their metrics too. 
+
+Be sure to access the linkerd dashboard by using the command `linkerd dashboard` before querying, as the metrics are configured to only return the last 30s by default.
```

Comment:
Because metrics only return the last 30s by default, if you'd like to see the metrics, make sure there is something happening on your cluster. The easiest way to do this is to run `linkerd dashboard` and see the traffic generated by viewing that page.